### ISSUE TITLE
Figure out a way to merge scene serializers

### ISSUE THREAD

**Issue by dwelle** [excalidraw/excalidraw#611]
As discussed in https://github.com/excalidraw/excalidraw/pull/583#issuecomment-579734116, currently we use different scene serializers when storing to `localStorage`, and when saving to backend or JSON (file).

This means there is potential for unintended divergence, such as when we decide to persist additional state keys and forget to modify both serializers. Note that there's already a divergence: when storing to `localStorage`, we don't store `version` which we do in the other cases.

At the same time, we cannot simply use the same serializer, because we don't want to store the exact the same state (such as scroll position, etc.) in every situation.

Thus, we should figure out a way to mitigate (or minimize) the potential for unintended divergence ()

The helpers in question are:

- [`saveToLocalStorage()`](https://github.com/excalidraw/excalidraw/blob/82717bcedad3f9aa5860513dcf3994372a9073fc/src/scene/data.ts#L301) - when storing to `localStorage`.
- [`serializeAsJSON()`](https://github.com/excalidraw/excalidraw/blob/82717bcedad3f9aa5860513dcf3994372a9073fc/src/scene/data.ts#L29) - for everything else.
   
One way is to use a single API, but supply a list of keys to *omit* (i.e. using a blacklist, which will minimize the likelihood of omitting keys added in the future).

**Comment by lipis** [comment]
We probably want to store the `name` and possible new property: `timestamp` to the `serializeAsJSON`

**Comment by dwelle** [comment]
@lipis btw, if you mean that `timestamp` from #485, then I meant it for the list of ids. But I agree we should store it in both, either way.

**Comment by lipis** [comment]
> then I meant it for the list of ids.

Taken from the local storage or from the actual backend? I'm confused..

**Comment by dwelle** [comment]
When we store the id in localStorage upon restoring the scene from backend: https://github.com/excalidraw/excalidraw/pull/583/files#diff-6d3d22779fda1f37f3f684229c94ebf6R332

**Comment by lipis** [comment]
Then that's an issue for the other PR and we can have it without timestamps in the first version.

**Comment by dwelle** [comment]
I don't think so actually. If we don't add the timestamp in the first pass, and we decide we need it, then we'll need another migration in the future. Best to store it now and remove it later, if needed.

**Comment by lipis** [comment]
Incremental improvements I would say (instead of removing later).. That's how Excalidraw was evolved so fast :)

**Comment by dwelle** [comment]
So you want to implement the migration instead of storing it now (since we're pretty sure we'll use it)?

**Comment by lipis** [comment]
I'm good either way.

**Comment by dwelle** [comment]
If you're good either way, then let's add the timestamps, because it'd fall on me to write and maintain the migrations...

**Comment by mtolmacs** [comment]
This seems to no longer be the case as of 2025, let's reopen if it was a hasty determination on my end.
